Latest Release PDF Hyperlink Broken

When downloading the latest release, [v.2.0](https://github.com/devsecfranklin/franklin-resume/tree/v.2.0), when the PDF is opened and selecting "click here" on the "latest version of the resume" it results in [an error when navigating to the URI](https://franklin-resume.herokuapp.com/): 


**Link Text**
```
Latest Version of this Resume: click here
```

**Browser Output**
```
There's nothing here, yet.
```

I see a broken link in the `README.md` and plenty of other places. 

```
franklin@chonk:~/workspace/consulting/franklin-resume$ go test -v ./test/...
=== RUN   TestBiographyIntegrity
=== RUN   TestBiographyIntegrity/FilePermissions
=== RUN   TestBiographyIntegrity/DraftPlaceholderScan
=== RUN   TestBiographyIntegrity/VerifyMainResumeIntegration
--- PASS: TestBiographyIntegrity (0.00s)
    --- PASS: TestBiographyIntegrity/FilePermissions (0.00s)
    --- PASS: TestBiographyIntegrity/DraftPlaceholderScan (0.00s)
    --- PASS: TestBiographyIntegrity/VerifyMainResumeIntegration (0.00s)
=== RUN   TestLinkIntegrity
=== RUN   TestLinkIntegrity/Link:https://docs.projectcalico.org/security/kubernetes-network-policy).
    link_test.go:71: Broken Link: https://docs.projectcalico.org/security/kubernetes-network-policy). returned status code 404
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/concepts/containers/images/)
    link_test.go:71: Broken Link: https://kubernetes.io/docs/concepts/containers/images/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/#backing-up-an-etcd-cluster)
=== RUN   TestLinkIntegrity/Link:https://zenodo.org/records/5918861
    link_test.go:65: Connection Error: Failed to reach https://zenodo.org/records/5918861: Get "https://zenodo.org/records/5918861": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
=== RUN   TestLinkIntegrity/Link:https://github.com/pale-shadow/game-skynet-minecraft
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/#backing-up-an-etcd-cluster),
=== RUN   TestLinkIntegrity/Link:https://g.dev/franklin
=== RUN   TestLinkIntegrity/Link:http://www.reuters.com/article/us-motorola-nokia-idUSTRE66I24P20100719)
    link_test.go:55: Security Risk: Insecure HTTP link found: http://www.reuters.com/article/us-motorola-nokia-idUSTRE66I24P20100719). Upgrade to HTTPS.
    link_test.go:71: Broken Link: http://www.reuters.com/article/us-motorola-nokia-idUSTRE66I24P20100719) returned status code 401
=== RUN   TestLinkIntegrity/Link:https://twitter.com/indybsides)
    link_test.go:71: Broken Link: https://twitter.com/indybsides) returned status code 403
=== RUN   TestLinkIntegrity/Link:https://orcid.org/0000-0003-4586-8500
=== RUN   TestLinkIntegrity/Link:https://www.motorolasolutions.com/en_us.html
=== RUN   TestLinkIntegrity/Link:https://github.com/devsecfranklin/franklin-resume/blob/ceb25740e386654e7e7e637c114389b814c6a308/resume/resume.pdf)
    link_test.go:71: Broken Link: https://github.com/devsecfranklin/franklin-resume/blob/ceb25740e386654e7e7e637c114389b814c6a308/resume/resume.pdf) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/configure-pod-container/security-context/)
    link_test.go:71: Broken Link: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/concepts/configuration/secret/)
    link_test.go:71: Broken Link: https://kubernetes.io/docs/concepts/configuration/secret/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://www.marketingcloud.com/)
=== RUN   TestLinkIntegrity/Link:https://trust.marketingcloud.com/trust]
=== RUN   TestLinkIntegrity/Link:https://en.wikipedia.org/wiki/Nokia_Networks).
    link_test.go:71: Broken Link: https://en.wikipedia.org/wiki/Nokia_Networks). returned status code 403
=== RUN   TestLinkIntegrity/Link:http://sps.northwestern.edu/program-areas/graduate/information-systems/)
    link_test.go:55: Security Risk: Insecure HTTP link found: http://sps.northwestern.edu/program-areas/graduate/information-systems/). Upgrade to HTTPS.
    link_test.go:71: Broken Link: http://sps.northwestern.edu/program-areas/graduate/information-systems/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://github.com/theDevilsVoice/iptables-spec-file-parser/blame/master/ip_tables-fileparser.sh)
    link_test.go:71: Broken Link: https://github.com/theDevilsVoice/iptables-spec-file-parser/blame/master/ip_tables-fileparser.sh) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://github.com/theDevilsVoice/milton/blob/master/summarize.py)
    link_test.go:71: Broken Link: https://github.com/theDevilsVoice/milton/blob/master/summarize.py) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://www.roosevelt.edu/CAS/Programs/CS.aspx).
    link_test.go:71: Broken Link: https://www.roosevelt.edu/CAS/Programs/CS.aspx). returned status code 404
=== RUN   TestLinkIntegrity/Link:http://www.hak4kidz.com/activities/grr.html)
    link_test.go:55: Security Risk: Insecure HTTP link found: http://www.hak4kidz.com/activities/grr.html). Upgrade to HTTPS.
=== RUN   TestLinkIntegrity/Link:https://twitch.tv/ibroadband
    link_test.go:71: Broken Link: https://twitch.tv/ibroadband returned status code 405
=== RUN   TestLinkIntegrity/Link:https://nsn.com
    link_test.go:65: Connection Error: Failed to reach https://nsn.com: Get "https://nsn.com": tls: failed to verify certificate: x509: certificate is valid for nokia.com, developer.nokia.com, www.alcatel-lucent.com.tr, alcatel-lucent.com.tr, dac.nokia.com, contenthub-origin-34599.nokia.com, contenthub.nokia.com, www.nokia.com, origin-prod-training.nokia.com, training.nokia.com, www.ftr.bz, ftr.bz, www.futurithmic.com, futurithmic.com, analytics.nokia.com, analytics-preview.nokia.com, mynokia.com, www.mynokia.com, www.apps.networks.nokia.com, apps.networks.nokia.com, ventures.nokia.com, support.ozo.nokia.com, bell-labs.com, www.bell-labs.com, not nsn.com
=== RUN   TestLinkIntegrity/Link:https://zenodo.org/records/19122412
    link_test.go:65: Connection Error: Failed to reach https://zenodo.org/records/19122412: Get "https://zenodo.org/records/19122412": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/concepts/policy/pod-security-policy/)
    link_test.go:71: Broken Link: https://kubernetes.io/docs/concepts/policy/pod-security-policy/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/)
    link_test.go:71: Broken Link: https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://github.com/theDevilsVoice/fubaria-chan/blob/master/python/channel.py)
    link_test.go:71: Broken Link: https://github.com/theDevilsVoice/fubaria-chan/blob/master/python/channel.py) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://www.youtube.com/channel/UC6xvUC5NThwShk-wrhh01Mg
=== RUN   TestLinkIntegrity/Link:https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml/badge.svg)](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml)
    link_test.go:71: Broken Link: https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml/badge.svg)](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://github.com/\github
    link_test.go:71: Broken Link: https://github.com/\github returned status code 404
=== RUN   TestLinkIntegrity/Link:https://github.com/theDevilsVoice/pystemon)
    link_test.go:71: Broken Link: https://github.com/theDevilsVoice/pystemon) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://github.com/theDevilsVoice/1337-Noms-The-Hacker-Cookbook)
    link_test.go:71: Broken Link: https://github.com/theDevilsVoice/1337-Noms-The-Hacker-Cookbook) returned status code 404
=== RUN   TestLinkIntegrity/Link:http://www.geni.net/)
    link_test.go:55: Security Risk: Insecure HTTP link found: http://www.geni.net/). Upgrade to HTTPS.
    link_test.go:71: Broken Link: http://www.geni.net/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://networks.nokia.com/)
    link_test.go:71: Broken Link: https://networks.nokia.com/) returned status code 403
=== RUN   TestLinkIntegrity/Link:http://www.cdm.depaul.edu/academics/Pages/MSinNetworkEngineeringandManagement.aspx)
    link_test.go:55: Security Risk: Insecure HTTP link found: http://www.cdm.depaul.edu/academics/Pages/MSinNetworkEngineeringandManagement.aspx). Upgrade to HTTPS.
    link_test.go:71: Broken Link: http://www.cdm.depaul.edu/academics/Pages/MSinNetworkEngineeringandManagement.aspx) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://www.paloaltonetworks.com/about-us
=== RUN   TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/)
    link_test.go:71: Broken Link: https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/) returned status code 404
=== RUN   TestLinkIntegrity/Link:https://www.linkedin.com/in/\LinkedIn
    link_test.go:71: Broken Link: https://www.linkedin.com/in/\LinkedIn returned status code 405
=== RUN   TestLinkIntegrity/Link:https://www.ucdenver.edu
=== RUN   TestLinkIntegrity/Link:http://odo.kettering.edu/dvipdfm/
    link_test.go:55: Security Risk: Insecure HTTP link found: http://odo.kettering.edu/dvipdfm/. Upgrade to HTTPS.
    link_test.go:65: Connection Error: Failed to reach http://odo.kettering.edu/dvipdfm/: Get "http://odo.kettering.edu/dvipdfm/": dial tcp: lookup odo.kettering.edu on 10.10.8.1:53: no such host
--- FAIL: TestLinkIntegrity (61.62s)
    --- FAIL: TestLinkIntegrity/Link:https://docs.projectcalico.org/security/kubernetes-network-policy). (0.74s)
    --- FAIL: TestLinkIntegrity/Link:https://kubernetes.io/docs/concepts/containers/images/) (0.18s)
    --- PASS: TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/#backing-up-an-etcd-cluster) (0.08s)
    --- FAIL: TestLinkIntegrity/Link:https://zenodo.org/records/5918861 (20.00s)
    --- PASS: TestLinkIntegrity/Link:https://github.com/pale-shadow/game-skynet-minecraft (0.94s)
    --- PASS: TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/#backing-up-an-etcd-cluster), (0.04s)
    --- PASS: TestLinkIntegrity/Link:https://g.dev/franklin (1.13s)
    --- FAIL: TestLinkIntegrity/Link:http://www.reuters.com/article/us-motorola-nokia-idUSTRE66I24P20100719) (1.33s)
    --- FAIL: TestLinkIntegrity/Link:https://twitter.com/indybsides) (0.22s)
    --- PASS: TestLinkIntegrity/Link:https://orcid.org/0000-0003-4586-8500 (0.23s)
    --- PASS: TestLinkIntegrity/Link:https://www.motorolasolutions.com/en_us.html (0.51s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/devsecfranklin/franklin-resume/blob/ceb25740e386654e7e7e637c114389b814c6a308/resume/resume.pdf) (0.31s)
    --- FAIL: TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) (0.07s)
    --- FAIL: TestLinkIntegrity/Link:https://kubernetes.io/docs/concepts/configuration/secret/) (0.06s)
    --- PASS: TestLinkIntegrity/Link:https://www.marketingcloud.com/) (4.91s)
    --- PASS: TestLinkIntegrity/Link:https://trust.marketingcloud.com/trust] (1.50s)
    --- FAIL: TestLinkIntegrity/Link:https://en.wikipedia.org/wiki/Nokia_Networks). (0.26s)
    --- FAIL: TestLinkIntegrity/Link:http://sps.northwestern.edu/program-areas/graduate/information-systems/) (0.29s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/theDevilsVoice/iptables-spec-file-parser/blame/master/ip_tables-fileparser.sh) (0.16s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/theDevilsVoice/milton/blob/master/summarize.py) (0.17s)
    --- FAIL: TestLinkIntegrity/Link:https://www.roosevelt.edu/CAS/Programs/CS.aspx). (0.65s)
    --- FAIL: TestLinkIntegrity/Link:http://www.hak4kidz.com/activities/grr.html) (0.24s)
    --- FAIL: TestLinkIntegrity/Link:https://twitch.tv/ibroadband (0.33s)
    --- FAIL: TestLinkIntegrity/Link:https://nsn.com (0.55s)
    --- FAIL: TestLinkIntegrity/Link:https://zenodo.org/records/19122412 (20.01s)
    --- FAIL: TestLinkIntegrity/Link:https://kubernetes.io/docs/concepts/policy/pod-security-policy/) (0.07s)
    --- FAIL: TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/) (0.06s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/theDevilsVoice/fubaria-chan/blob/master/python/channel.py) (0.16s)
    --- PASS: TestLinkIntegrity/Link:https://www.youtube.com/channel/UC6xvUC5NThwShk-wrhh01Mg (0.32s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml/badge.svg)](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml) (0.19s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/\github (0.17s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/theDevilsVoice/pystemon) (0.19s)
    --- FAIL: TestLinkIntegrity/Link:https://github.com/theDevilsVoice/1337-Noms-The-Hacker-Cookbook) (0.16s)
    --- FAIL: TestLinkIntegrity/Link:http://www.geni.net/) (0.23s)
    --- FAIL: TestLinkIntegrity/Link:https://networks.nokia.com/) (0.64s)
    --- FAIL: TestLinkIntegrity/Link:http://www.cdm.depaul.edu/academics/Pages/MSinNetworkEngineeringandManagement.aspx) (2.15s)
    --- PASS: TestLinkIntegrity/Link:https://www.paloaltonetworks.com/about-us (1.60s)
    --- FAIL: TestLinkIntegrity/Link:https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/) (0.07s)
    --- FAIL: TestLinkIntegrity/Link:https://www.linkedin.com/in/\LinkedIn (0.19s)
    --- PASS: TestLinkIntegrity/Link:https://www.ucdenver.edu (0.21s)
    --- FAIL: TestLinkIntegrity/Link:http://odo.kettering.edu/dvipdfm/ (0.28s)
FAIL
FAIL    github.com/devsecfranklin/franklin-resume/test  61.626s
FAIL
```

--- a/.github/workflows/security-scan.yml
+++ b/.github/workflows/security-scan.yml
@@ -1,0 +1,30 @@
+name: Security & Integrity Scan
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  go-security-tests:
+    name: Run Asset & Link Integrity Checks
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Go Environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+          cache: true # Slickly caches dependencies for faster runs
+
+      - name: Initialize Go Modules
+        # Ensures the environment is ready even if go.sum is missing
+        run: go mod tidy
+
+      - name: Run Integrity Tests
+        # Executes your link_test.go and asset_test.go [cite: 63, 68]
+        run: go test -v ./test/...

--- a/README.md
+++ b/README.md
@@ -1,6 +1,8 @@
 # Franklin Diaz - Resume
 
-[![LaTeX build](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml/badge.svg)](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml)
+[![GitHub](https://img.shields.io/badge/GitHub-devsecfranklin-blue?logo=github)](https://github.com/devsecfranklin)
+[![ORCID](https://img.shields.io/badge/ORCID-0000--0003--4586--8500-A6CE39?logo=orcid&logoColor=FFFFFF)](https://orcid.org/0000-0003-4586-8500)
+[![LaTeX build](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml/badge.svg)](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml) [![Security & Integrity Scan](https://github.com/devsecfranklin/franklin-resume/actions/workflows/security-scan.yml/badge.svg)](https://github.com/devsecfranklin/franklin-resume/actions/workflows/security-scan.yml)
 
 - [here is the PDF file](https://github.com/devsecfranklin/franklin-resume/blob/main/resume/resume.pdf)
 

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 
 [![LaTeX build](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml/badge.svg)](https://github.com/devsecfranklin/franklin-resume/actions/workflows/latex.yaml)
 
-- [here is the PDF file](https://github.com/devsecfranklin/franklin-resume/blob/ceb25740e386654e7e7e637c114389b814c6a308/resume/resume.pdf)
+- [here is the PDF file](https://github.com/devsecfranklin/franklin-resume/blob/main/resume/resume.pdf)
 
 ## Project Overview
 

--- a/go.mod
+++ b/go.mod
@@ -1,0 +1,3 @@
+module github.com/devsecfranklin/franklin-resume
+
+go 1.26.3

--- a/test/link_test.go
+++ b/test/link_test.go
@@ -1,0 +1,75 @@
+package test
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestLinkIntegrity(t *testing.T) {
+	linkRegex := regexp.MustCompile(`https?://[^\s{}<>"]+`)
+	uniqueLinks := make(map[string]bool)
+	
+	// Corrected signature: must include info and err arguments
+	err := filepath.Walk("..", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// Only scan source/documentation files [cite: 283, 288]
+		if !strings.HasSuffix(path, ".tex") && !strings.HasSuffix(path, ".md") {
+			return nil
+		}
+		// Skip hidden directories like .git [cite: 277]
+		if strings.Contains(path, "/.") {
+			return nil
+		}
+
+		content, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		matches := linkRegex.FindAllString(string(content), -1)
+		for _, link := range matches {
+			cleanLink := strings.TrimRight(link, "}")
+			uniqueLinks[cleanLink] = true
+		}
+		return nil
+	})
+
+	if err != nil {
+		t.Fatalf("Failed to crawl repository for links: %v", err)
+	}
+
+	client := &http.Client{Timeout: 10 * time.Second}
+
+	for link := range uniqueLinks {
+		t.Run(fmt.Sprintf("Link:%s", link), func(t *testing.T) {
+			// Security Check: Enforce HTTPS [cite: 283, 287]
+			if strings.HasPrefix(link, "http://") {
+				t.Errorf("Security Risk: Insecure HTTP link found: %s. Upgrade to HTTPS.", link)
+			}
+
+			// Liveness Check: Ensure the link isn't broken 
+			resp, err := client.Head(link)
+			if err != nil {
+				resp, err = client.Get(link)
+			}
+
+			if err != nil {
+				t.Errorf("Connection Error: Failed to reach %s: %v", link, err)
+				return
+			}
+			defer resp.Body.Close()
+
+			if resp.StatusCode >= 400 {
+				t.Errorf("Broken Link: %s returned status code %d", link, resp.StatusCode)
+			}
+		})
+	}
+}

